site/en/blog/canvas2d/index.md

```diff
@@ -0,0 +1,282 @@
+  ctx.stroke();
+```
+
+{% Img src="image/kheDArv5csY6rvQUJDbWRscckLr1/VlZkoPMTOuzD6bUeTd1i.png", alt="A horizontal and a diagonal line.", width="500", height="500" %}
+
+Grrrr! That's not right! 😡 What's that extra line doing here? Also, why is it pink? Okay, let's just check StackOverflow.
+
+```js
+canvas.width = canvas.width;
+```
+
+Why is this so silly? Why is this so hard?
+
+Well, it's not any more. With the new api we have the simple, elegant, beautiful groundbreaking:
+
+```js
+ctx.reset();
+```
+
+Sorry that took so long.
+
+### Filters
+
+SVG filters are a world unto themselves. If they're new to you I highly recommend reading "[The Art Of SVG Filters And Why It Is Awesome](https://www.smashingmagazine.com/2015/05/why-the-svg-filter-is-awesome/)", which shows some of their amazing potential.
+
+SVG style filters are already available for Canvas2D! You just have to be willing to pass the filter as a url pointing to another SVG filter element on the page: 
+
+```html
+<svg>
+    <defs>
+     <filter id="svgFilter">
+       <feGaussianBlur in="SourceGraphic" stdDeviation="5"/>
+       <feConvolveMatrix kernelMatrix="-3 0 0 0 0.5 0 0 0 3"/>
+       <feColorMatrix type="hueRotate" values="90"/>
+     </filter>
+    </defs>
+ </svg>
+ ```
+
+```js
+  const canvas = document.createElement("canvas");
+  canvas.width = 500;
+  canvas.height = 400;
+  const ctx = canvas.getContext("2d");
+  document.body.appendChild(canvas);
+
+  ctx.filter = "url('#svgFilter')";
+  draw90sPattern(ctx);
+```
+
+Which messes up our pattern pretty good:
+
+{% Img src="image/kheDArv5csY6rvQUJDbWRscckLr1/IPgddXUAwJFZzowzdxtx.png", alt="The retro pattern with a blurred effect applied.", width="500", height="500" %}
+
+But, what if you wanted to do the above but stay within JavaScript and not mess around with strings? With the new API, this is totally possible.
+
+```js
+  ctx.filter = new CanvasFilter([
+    {filter: "gaussianBlur", stdDeviation: 5},
+    {filter: "convolveMatrix",
+      kernelMatrix: [
+        [-3, 0, 0],
+        [0, 0.5, 0],
+        [0, 0, 3]]
+    },
+    {filter: "colorMatrix", type: "hueRotate", values: 90},
+  ]);
+```
+
+Easy as pie! Try it out with and play with the parameters [in the demo here](https://glitch.com/edit/#!/pretty-dent-titanoceratops).
+
+## Part 3: Performance Improvements
+
+With the New Canvas2D API, we also wanted to improve performance where possible. We added a couple features to give developers finer-grained control of their websites and allow for the slickest possible framerates:
+
+### Will Read Frequently
+
+getImageData() is how you read pixel data back from a canvas. It can be very slow. The new API gives you a way of explicitly marking a canvas for readback (for generative effects, for example). This allows you to optimize things under the hood and keep canvas fast for a larger variety of use cases. This feature has been around in Firefox for a while and we're finally making it part of the canvas spec.
```

Comment:
```suggestion
Use `getImageData()` to read pixel data back from a canvas. It can be very slow. The new API gives you a way of explicitly marking a canvas for reading back (for generative effects, for example). This allows you to optimize things under the hood and keep canvas fast for a larger variety of use cases. This feature has been in Firefox for a while and we're finally making it part of the canvas spec.
```